### DISCUSSION TITLE
ReactiveUI Samples for SingleViewApplicationLifetime and compiled bindings

### DISCUSSION THREAD

**Discussion by lylerolleman** [AvaloniaUI/Avalonia#12633] [General]
I've been trying to follow the ReactiveUI deep dive with a new AvaloniaUI 11 cross platform app but I've been running into issues, notably that it seems to assume only desktop applications (IClassicDesktopStyleApplicationLifetime), -and it doesn't seem to work with compiled bindings (cannot infer the view model, even when I specify it in the XAML)- nevermind compiled bindings, they work fine, just mixed up the x:DataType

Are there any up to date samples on how to do this? I could use the community toolkit version, but that doesn't really provide you much to work with (doesn't even seem to have any kind of navigation...)

**Comment by thevortexcloud** [comment]
The process is really not that much different. But instead of having multiple windows you just shove everything inside a single view. How you do that is up to you. A fairly simple approach that works well for small apps is to just hide/show pages as needed via some sort of display manager. EG put all your view models/controls into a single view and then bind their visibility to something managed by the main view view model.
